src/EnergyPlus/HVACManager.cc

```diff
@@ -2725,6 +2730,46 @@ namespace HVACManager {
 
 	}
 
+	void
+	CheckMassBalances()
+	{
+		using namespace Fans;
+		using namespace Pumps;
+		using General::RoundSigDigits;
+		using DataHVACGlobals::SmallMassFlow;  // Air
+		using DataBranchAirLoopPlant::MassFlowTolerance;  // Liquid
+
+		for (int FanNum = 1; FanNum <= NumFans; ++FanNum) {
+
+//			if (abs(Node( Fan( FanNum ).InletNodeNum ).MassFlowRate - Fan( FanNum ).InletAirMassFlowRate) > SmallMassFlow) {
+//				ShowFatalError(Fan(FanNum).FanType + ", Name: " + Fan(FanNum).FanName + ": Fan inlet mass flow rate (" + RoundSigDigits(Fan( FanNum ).InletAirMassFlowRate, 4) + ") does not match inlet node mass flow rate (" + RoundSigDigits(Node( Fan( FanNum ).InletNodeNum ).MassFlowRate, 4) + ").");
+//			}
+//
+//			if (abs(Node( Fan( FanNum ).OutletNodeNum ).MassFlowRate - Fan( FanNum ).OutletAirMassFlowRate) > SmallMassFlow) {
+//				ShowFatalError(Fan(FanNum).FanType + ", Name: " + Fan(FanNum).FanName + ": Fan outlet mass flow rate (" + RoundSigDigits(Fan( FanNum ).OutletAirMassFlowRate, 4) + ") does not match outlet node mass flow rate (" + RoundSigDigits(Node( Fan( FanNum ).OutletNodeNum ).MassFlowRate, 4) + ").");
+//			}
+
+			if (abs(Node( Fan( FanNum ).InletNodeNum ).MassFlowRate - Node( Fan( FanNum ).OutletNodeNum ).MassFlowRate) > SmallMassFlow) {
+//				ShowFatalError(Fan(FanNum).FanType + ", Name: " + Fan(FanNum).FanName + ": Fan outlet mass flow rate (" + RoundSigDigits(Node( Fan( FanNum ).OutletNodeNum ).MassFlowRate, 4) + ") does not match fan inlet node mass flow rate (" + RoundSigDigits(Node( Fan( FanNum ).InletNodeNum ).MassFlowRate, 4) + ").");
+// can we get a handle on why this is happening before we start crashing user files?
```

Comment:
That's where I have it right now. Pump does not have separate inlet/outlet flow rate variables that I'm aware of. 

It would be good to do this around other components as well (e.g. coils, HXs, mixers, splitters) just to be sure. Zones would be good, too. These checks are relatively inexpensive if done in the right place. If anything like this pops up elsewhere, it would be good to have a way to find out about it.